dotnet/src/SemanticKernel.Abstractions/Orchestration/SKContext.cs

```diff
@@ -64,57 +64,25 @@ public CultureInfo Culture
     /// The kernel context is a lightweight instance of the main kernel with its services.
     /// </remarks>
     /// <returns>Kernel reference</returns>
-    private IKernel GetKernelContext()
-        => this._originalKernel; // TODO: Clone a lightweight kernel instead of returning the same instance
+    private IKernelExecutionContext GetKernelContext()
+        => (IKernelExecutionContext)this._kernelContext; // TODO: Clone a lightweight kernel instead of returning the same instance
 
     /// <summary>
     /// Constructor for the context.
     /// </summary>
-    /// <param name="kernel">Kernel reference</param>
+    /// <param name="kernelContext">Kernel reference</param>
     /// <param name="variables">Context variables to include in context.</param>
-    /// <param name="functions">Functions to include in context.</param>
-    public SKContext(
-        IKernel kernel,
+    /// <param name="culture">Culture related to the context</param>
+    internal SKContext(
+        IKernelExecutionContext kernelContext,
```

Comment:
We can keep the `IKernel kernel` parameter but return the kernel as an instance of the IKernelExecutionContext from the `Kernel` property getter. It'll allow to avoid putting the LoggerFactory and Functions into the IKernelExecutionContext interface.

Reply:
Made the change following the discussion regarding `IFunctionExecutor`